tests/models/superpoint/test_modeling_superpoint.py

```diff
@@ -0,0 +1,274 @@
+    )
+
+if is_vision_available():
+    from PIL import Image
+
+    from transformers import AutoImageProcessor
+
+
+class SuperPointModelTester:
+    def __init__(
+        self,
+        parent,
+        batch_size=3,
+        image_width=640,
+        image_height=480,
+        conv_layers_sizes: List[int] = [64, 64, 128, 128, 256],
+        descriptor_dim: int = 256,
+        keypoint_threshold: float = 0.005,
+        max_keypoints: int = -1,
+        nms_radius: int = 4,
+        border_removal_distance: int = 4,
+    ):
+        self.parent = parent
+        self.batch_size = batch_size
+        self.image_width = image_width
+        self.image_height = image_height
+        self.conv_layers_sizes = conv_layers_sizes
+        self.descriptor_dim = descriptor_dim
+        self.keypoint_threshold = keypoint_threshold
+        self.max_keypoints = max_keypoints
+        self.nms_radius = nms_radius
+        self.border_removal_distance = border_removal_distance
+
+    def prepare_config_and_inputs(self):
+        # SuperPoint expects a grayscale image as input
+        pixel_values = floats_tensor([self.batch_size, 3, self.image_width, self.image_height])
+        config = self.get_config()
+        return config, pixel_values
+
+    def get_config(self):
+        return SuperPointConfig(
+            conv_layers_sizes=self.conv_layers_sizes,
+            descriptor_dim=self.descriptor_dim,
+            keypoint_threshold=self.keypoint_threshold,
+            max_keypoints=self.max_keypoints,
+            nms_radius=self.nms_radius,
+            border_removal_distance=self.border_removal_distance,
+        )
+
+    def create_and_check_model(self, config, pixel_values):
+        model = SuperPointModel(config=config)
+        model.to(torch_device)
+        model.eval()
+        result = model(pixel_values)
+        self.parent.assertEqual(
+            result.last_hidden_state.shape,
+            (self.batch_size, self.conv_layers_sizes[-2], self.image_width // 8, self.image_height // 8),
+        )
+
+    def create_and_check_for_interest_point_description(self, config, pixel_values):
+        model = SuperPointModelForInterestPointDescription(config=config)
+        model.to(torch_device)
+        model.eval()
+        result = model(pixel_values)
+
+        self.parent.assertEqual(
+            result.last_hidden_state.shape,
+            (self.batch_size, self.hidden_sizes[-1], self.image_size // 32, self.image_size // 32),
+        )
+
+    def prepare_config_and_inputs_for_common(self):
+        config_and_inputs = self.prepare_config_and_inputs()
+        config, pixel_values = config_and_inputs
+        inputs_dict = {"pixel_values": pixel_values}
+        return config, inputs_dict
+
+
+@require_torch
+class SuperPointModelTest(ModelTesterMixin, unittest.TestCase):
+    all_model_classes = (SuperPointModel, SuperPointModelForInterestPointDescription) if is_torch_available() else ()
```

Comment:
`SuperPointModelForInterestPointDescription` has to be renamed to `SuperPointForInterestPointDescription`

Reply:
Fixed in [this commit](https://github.com/huggingface/transformers/pull/25786/commits/a8f7e1c97b89d563552c0e32c350eb7ebead1506)